perun-core/src/main/java/cz/metacentrum/perun/core/impl/AuditMessagesManagerImpl.java

```diff
@@ -214,12 +219,16 @@ public List<AuditMessage> getMessagesByIdAndCount(PerunSession perunSession, int
 
 	@Override
 	public Paginated<AuditMessage> getMessagesPage(PerunSession perunSession, MessagesPageQuery query) {
+		MapSqlParameterSource parameters = new MapSqlParameterSource();
+		parameters.addValue("selectedEvents", query.getSelectedEvents());
+
 		// take exact total count up to PAGE_COUNT_PRECISION entries, estimate it otherwise
-		return jdbc.query("select " + auditMessageMappingSelectQuery
-			+ ", case when (select count(*) from (select 1 from auditer_log limit" + PAGE_COUNT_PRECISION + ") as sample) < " + PAGE_COUNT_PRECISION + " then (select count(*) from auditer_log) else " +
-			"(select 100 * count(*) FROM auditer_log TABLESAMPLE SYSTEM (1)) end as total_count "
-			+ "from auditer_log order by id " + query.getOrder().getSqlValue() + " offset " + query.getOffset()
-			+ " limit " + query.getPageSize(), getPaginatedMessagesExtractor(query));
+		return namedParameterJdbcTemplate.query("select " + auditMessageMappingSelectQuery
+			+ ", case when (select count(*) from (select 1 from auditer_log " + (!query.getSelectedEvents().isEmpty() ? "where split_part(msg::json ->>'name', '.', 7) IN (:selectedEvents) " : "") + "limit " + PAGE_COUNT_PRECISION + ") as sample) < " + PAGE_COUNT_PRECISION
+			+ " then (select count(*) from auditer_log" + (!query.getSelectedEvents().isEmpty() ? " where split_part(msg::json ->>'name', '.', 7) IN (:selectedEvents)" : "") +")"
+			+ " else (select 100 * count(*) FROM auditer_log TABLESAMPLE SYSTEM (1)) end as total_count "
```

Comment:
```suggestion
			+ " else (select 100 * count(*) FROM auditer_log TABLESAMPLE SYSTEM (1)" + (!query.getSelectedEvents().isEmpty() ? " where split_part(msg::json ->>'name', '.', 7) IN (:selectedEvents)" : "") +") end as total_count "
```
the 1% sample should also be filtered by the event name, otherwise the estimation of the total count is again for the unfiltered table.